Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Material/MaterialAsset.h

```diff
@@ -126,6 +127,9 @@ namespace AZ
             //! Holds values for each material property, used to initialize Material instances.
             //! This is indexed by MaterialPropertyIndex and aligns with entries in m_materialPropertiesLayout.
             AZStd::vector<MaterialPropertyValue> m_propertyValues;
+            //! This is used to find the MaterialPropertyIndex from materialTypeAsset.m_materialPropertiesLayout
+            //! to match to the appropriate shader index and property value. 
+            AZStd::vector<AZ::Name> m_propertyNames;
```

Comment:
The comment should probably mention that this list could be empty, and what that implies.